### ISSUE TITLE
Rule Proposal: Boolean prop shorthand (`prop` instead of `:prop="true"`)

### ISSUE THREAD

**Issue by manniL** [vuejs/eslint-plugin-vue#775]
**Please describe what the rule should do:**

I've seen code like `<div :boolProp="true" />` often where you could use `<div boolProp />` instead. I'd love to see a rule for that (plus an auto-fix option which should be fairly possible)

**What category should the rule belong to?**
- [x] Enforces code style
- [ ] Warns about a potential error
- [x] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<template>
  <div>
    <button :disabled="true">
  </div>
</template>
```

**Comment by armano2** [comment]
i think there should be an option to enable/disable this feature.
- `always`
```vue
// valid
<button :disabled="true">
// invalid
<button disabled>
  ```
- `never`
```vue
// valid
<button disabled>
// invalid
<button :disabled="true">
```

**Comment by manniL** [comment]
@armano2 Fully agree. `always`/`never` should be configurable.

Like the name :relaxed:
